File: single-language-analysis/dataset-files/java/TestSmell157.java
import org.junit.Test;
import static org.junit.Assert.*;
import java.util.concurrent.TimeUnit;

public class TestSmell157 {

    @Test
    public void testPriorityOrdering() {
        // First request with MEDIUM priority
        Response response1 = client.post(
            "/ingest",
            new JsonObject().add("ids", new JsonArray().add(1).add(2).add(3).add(4).add(5)).add("priority", "MEDIUM")
        );
        String ingestionId1 = response1.json().getString("ingestion_id");
        
        // Second request with HIGH priority
        Response response2 = client.post(
            "/ingest",
            new JsonObject().add("ids", new JsonArray().add(6).add(7).add(8).add(9)).add("priority", "HIGH")
        );
        String ingestionId2 = response2.json().getString("ingestion_id");
        
        // Wait for some processing
        try {
            Thread.sleep(6);
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
        
        // Check status of both requests
        Response status1 = client.get("/status/" + ingestionId1);
        Response status2 = client.get("/status/" + ingestionId2);
        
        // HIGH priority request should have more completed batches
        int completedBatches1 = 0;
        for (JsonObject batch : status1.json().getJsonArray("batches").getValuesAs(JsonObject.class)) {
            if (batch.getString("status").equals(BatchStatus.COMPLETED)) {
                completedBatches1++;
            }
        }
        
        int completedBatches2 = 0;
        for (JsonObject batch : status2.json().getJsonArray("batches").getValuesAs(JsonObject.class)) {
            if (batch.getString("status").equals(BatchStatus.COMPLETED)) {
                completedBatches2++;
            }
        }
        
        assertEquals(true, completedBatches2 >= completedBatches1);
    }
}

File: single-language-analysis/dataset-files/python/test_smell_157.py
# Original URL: https://github.com/bhuvesh-bansal/LoopAiApi/blob/e9de96fcb2d1e272ee9fc59fe2158cbacf3a88db/test_main.py#L29-L57

def test_smell_157():
    # First request with MEDIUM priority
    response1 = client.post(
        "/ingest",
        json={"ids": [1, 2, 3, 4, 5], "priority": "MEDIUM"}
    )
    ingestion_id1 = response1.json()["ingestion_id"]
    
    # Second request with HIGH priority
    response2 = client.post(
        "/ingest",
        json={"ids": [6, 7, 8, 9], "priority": "HIGH"}
    )
    ingestion_id2 = response2.json()["ingestion_id"]
    
    # Wait for some processing
    time.sleep(6)
    
    # Check status of both requests
    status1 = client.get(f"/status/{ingestion_id1}")
    status2 = client.get(f"/status/{ingestion_id2}")
    
    # HIGH priority request should have more completed batches
    completed_batches1 = sum(1 for batch in status1.json()["batches"] 
                           if batch["status"] == BatchStatus.COMPLETED)
    completed_batches2 = sum(1 for batch in status2.json()["batches"] 
                           if batch["status"] == BatchStatus.COMPLETED)
    
    assert completed_batches2 >= completed_batches1